src/modules/gnd_pos_control/gnd_pos_control_main.cpp

```diff
@@ -0,0 +1,1405 @@
+	float	_althold_epv;				/**< the position estimate accuracy when engaging alt hold */
+	struct position_setpoint_s _hdg_hold_prev_wp;	/**< position where heading hold started */
+	struct position_setpoint_s _hdg_hold_curr_wp;	/**< position to which heading hold flies */
+	hrt_abstime _control_position_last_called; 	/**<last call of control_position  */
+
+	/* Takeoff launch detection and runway */
+	LaunchDetectionResult _launch_detection_state;
+
+	/* Pid controller for the speed. Here we assume we can control airspeed but the control variable is actually on
+	 the throttle. For now just assuming a proportional scaler between controlled airspeed and throttle output.*/
+	PID_t _speed_ctrl;
+
+	/* throttle and airspeed states */
+	float _speed_error;				///< airspeed error to setpoint in m/s
+	bool _airspeed_valid;				///< flag if a valid airspeed estimate exists
+	uint64_t _airspeed_last_received;		///< last time airspeed was received. Used to detect timeouts.
+		
+	bool _global_pos_valid;				///< global position is valid
+	math::Matrix<3, 3> _R_nb;			///< current attitude
+	float _roll;
+	float _pitch;
+	float _yaw;
+	bool _reinitialize_tecs;			///< indicates if the TECS states should be reinitialized (used for VTOL)
+	bool _is_tecs_running;
+	hrt_abstime _last_tecs_update;
+	float _asp_after_transition;
+	bool _was_in_transition;
+
+	// estimator reset counters
+	uint8_t _pos_reset_counter;		// captures the number of times the estimator has reset the horizontal position
+	uint8_t _alt_reset_counter;		// captures the number of times the estimator has reset the altitude state
+
+	ECL_L1_Pos_Controller				_gnd_control;
+	TECS						_tecs;
+	enum FW_POSCTRL_MODE {
+		FW_POSCTRL_MODE_AUTO,
+		FW_POSCTRL_MODE_POSITION,
+		FW_POSCTRL_MODE_ALTITUDE,
+		FW_POSCTRL_MODE_OTHER
+	} _control_mode_current;			///< used to check the mode in the last control loop iteration. Use to check if the last iteration was in the same mode.
+
+	struct {
+		float l1_period;
+		float l1_damping;
+		float l1_distance;
+
+		float time_const;
+		float time_const_throt;
+		float min_sink_rate;
+		float max_sink_rate;
+		float max_climb_rate;
+		float heightrate_p;
+		float heightrate_ff;
+		float speedrate_p;
+		float throttle_damp;
+		float integrator_gain;
+		float vertical_accel_limit;
+		float height_comp_filter_omega;
+		float speed_comp_filter_omega;
+		float roll_throttle_compensation;
+		float speed_weight;
+		float pitch_damping;
+
+		float airspeed_min;
+		float airspeed_trim;
+		float airspeed_max;
+		int airspeed_mode;
+		int speed_control_mode;
+		float speed_p;
+		float speed_i;
+		float speed_d;
+		float speed_imax;
+		float speed_throttle_airspeed_scaler;
+
+		float pitch_limit_min;
+		float pitch_limit_max;
+		float roll_limit;
+		float throttle_min;
+		float throttle_max;
+		float throttle_cruise;
+		float throttle_slew_max;
+		float man_roll_max_rad;
+		float man_pitch_max_rad;
+
+
+	} _parameters;			/**< local copies of interesting parameters */
```

Comment:
you can leave some but others like `float airspeed_min` don't make sense. Sink and climb are for fw imo.